sup3r/pipeline/forward_pass.py

```diff
@@ -1705,24 +1728,36 @@ def incremental_check_run(cls, strategy, node_index, chunk_index):
             corresponding file set, cropped_file_slice, padded_file_slice,
             and padded/overlapping/cropped spatial slice for a spatiotemporal
             chunk
+
+        Returns
+        -------
+        ForwardPass | None
+            If the forward pass for the given chunk is not finished this
+            returns an initialized forward pass object, otherwise returns None
         """
-        out_file = strategy.out_files[chunk_index]
-        if os.path.exists(out_file) and strategy.incremental:
-            logger.info('Not running chunk index {}, output file '
-                        'exists: {}'.format(chunk_index, out_file))
-        else:
-            try:
-                fwp = cls(strategy, chunk_index, node_index)
-                logger.info(f'Running forward pass for '
-                            f'chunk_index={chunk_index}, '
-                            f'node_index={node_index}, '
-                            f'file_paths={fwp.file_paths}')
+        fwp = None
+        if not strategy.chunk_finished(chunk_index):
+            fwp = cls(strategy, chunk_index=chunk_index, node_index=node_index)
+        return fwp
+
+    @classmethod
+    def _run_fwp_or_future(cls, fwp):
+        """Check if fwp is ForwardPass object or future and start forward
+        pass
+
+        Parameters
+        ----------
+        fwp : ForwardPass | future | None
+            Either a ForwardPass object, a concurrent future, or None. This
+            comes from either a direct ForwardPass initialization or a
+            ThreadPoolExecutor.submit() call.
+        """
+        if isinstance(fwp, ForwardPass):
+            fwp.run_chunk()
+        elif fwp is not None:
+            fwp = fwp.result()
+            if fwp is not None:
                 fwp.run_chunk()
```

Comment:
This is some weird double "if fwp is not None" logic... What do you think about this?
```
if isinstance(fwp, future):  # need to find future class from cf
    fwp = fwp.result()
if isinstance(fwp, ForwardPass):
    fwp.run_chunk()

Reply:
Yeah thats better